upload/uploader.go

```diff
@@ -86,7 +88,25 @@ func (u Uploader) Upload() error {
 		return errors.WithStack(err)
 	}
 
-	return u.SendBatches(testReport, batchLinks.Links.PostBatch)
+	postBatchURL := u.TransformPostBatchURL(batchLinks.Links.PostBatch)
+
+	return u.SendBatches(testReport, postBatchURL)
+}
+
+func (u Uploader) TransformPostBatchURL(rawUrl string) string {
+	if !u.Insecure {
+		return rawUrl
+	}
+
+	parsed, err := url.Parse(rawUrl)
+
+	if err != nil {
+		return rawUrl
+	}
+
+	parsed.Scheme = "http"
+
+	return parsed.String()
```

Comment:
Should we always set the scheme on the parsed url? https normally but if the insecure flag is set, http?

Reply:
Ah, interesting idea. That _would_ make the code simpler (we could remove the guard clause). 

That change might give the impression to readers, though, that the scheme we're receiving from the API is invalid or missing. It's not, really—we're just overriding it this very specific mode of operation.

I could go either way. 😄 Do you favor one way or the other?

Reply:
I kind of like that regardless of the URL scheme returned via the API, that we would be enforcing secure transport by default and only disabling it if the insecure flag is used.

Up to you though. Fine with whichever you prefer!

Reply:
> I kind of like that regardless of the URL scheme returned via the API, that we would be enforcing secure transport by default and only disabling it if the insecure flag is used.

I dig it! I'll make that change. 👍